### PR TITLE
HDDS-7408. Recon: Refactor container status page

### PR DESCRIPTION
**Pull request by smitajoshi12**
## What changes were proposed in this pull request?
Refractoring of Code for Missing Container End Point and changes suggested in JIRA.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7408

## How was this patch tested?
Manually Tested

### CODE REVIEW

**Comment by kerneltime** [comment]
@GeorgeJahad

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx [hunk 1]
```diff
@@ -244,32 +229,30 @@ export class MissingContainers extends React.Component<Record<string, object>, I
...
-      axios.get('/api/v1/containers/unhealthy/MISSING'),
-      axios.get('/api/v1/containers/unhealthy/UNDER_REPLICATED'),
-      axios.get('/api/v1/containers/unhealthy/OVER_REPLICATED'),
-      axios.get('/api/v1/containers/unhealthy/MIS_REPLICATED')
-    ]).then(axios.spread((missingContainersResponse, underReplicatedResponse, overReplicatedResponse, misReplicatedResponse, allReplicatedResponse) => {
+      axios.get('/api/v1/containers/unhealthy')
+    ]).then(axios.spread((allReplicatedResponse) => {
+
+      const allContainersResponseData: IUnhealthyContainersResponse = allReplicatedResponse.data;
+      const allContainers: IContainerResponse[] = allContainersResponseData.containers;
```
Comment (DaveTeng0) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx:236]: just curious~ what does 'I' in IContainerResponse mean?

Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx:236]: Interface

**Comment by GeorgeJahad** [comment]
LGTM, (but I should mention I haven't written much typescript.)

**Comment by kerneltime** [comment]
@smitajoshi12 would it make sense to include an image of the final output?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json [hunk 2]
```diff
@@ -24,8 +24,5 @@
   "/namespace/quota?path=*": "/quota",
   "/task/status": "/taskStatus",
   "/containers/missing": "/missingContainers",
```
Comment (dombizita) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json:26]: Please remove this endpoint, as it is not used anymore.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json [hunk 3]
```diff
@@ -24,8 +24,5 @@
...
-  "/containers/unhealthy/MISSING": "/containerUnhealthyMissing",
-  "/containers/unhealthy/UNDER_REPLICATED": "/containerUnhealthyUnderReplicated",
-  "/containers/unhealthy/OVER_REPLICATED": "/containerUnhealthyOverReplicated",
-  "/containers/unhealthy/MIS_REPLICATED": "/containerUnhealthyMisReplicated"
+  "/containers/unhealthy": "/containerUnhealthyResponse"
```
Comment (dombizita) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json:27]:
```suggestion
  "/containers/unhealthy": "/unhealthyContainers"
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx [hunk 4]
```diff
@@ -244,32 +229,30 @@ export class MissingContainers extends React.Component<Record<string, object>, I
     });
 
     axios.all([
-      axios.get('/api/v1/containers/unhealthy/MISSING'),
-      axios.get('/api/v1/containers/unhealthy/UNDER_REPLICATED'),
-      axios.get('/api/v1/containers/unhealthy/OVER_REPLICATED'),
-      axios.get('/api/v1/containers/unhealthy/MIS_REPLICATED')
-    ]).then(axios.spread((missingContainersResponse, underReplicatedResponse, overReplicatedResponse, misReplicatedResponse, allReplicatedResponse) => {
+      axios.get('/api/v1/containers/unhealthy')
+    ]).then(axios.spread((allReplicatedResponse) => {
```
Comment (dombizita) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx]:
```suggestion
    ]).then(axios.spread((allContainersResponse) => {
```

**Review by dombizita — COMMENTED** [review]
Thanks for the refactoring @smitajoshi12, overall it looks to me, I left a few inline comments.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json (hunk 2 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json:26]: In UI Overview.tsx we are using this endpoint.

Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json:26]: image

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json (hunk 3 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json]: Done Changes

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx (hunk 4 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/missingContainers/missingContainers.tsx]: Done Changes

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json (hunk 2 above)
Comment (dombizita) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json:26]: If we use the `/containers/missing` endpoint there why do we need to specify the `"/missingContainers"` route here? What would happen in the `overview.tsx` if we remove this from the `routes.json`?
```suggestion
```

Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/routes.json:26]: It is  key value pair  /missingContainer is associated  name for route /conatiner/missing we have to mention route in route.json. If we remove from overview.tsx code will break and UI will not work.

**Review by smengcl — APPROVED** [review]

**Comment by smengcl** [comment]
Thanks @smitajoshi12 for the patch. Thanks @GeorgeJahad and @dombizita for the review.

**Pull request merged**